### PR TITLE
tools(changelog): handle non merge commits

### PR DESCRIPTION
**Pull request by erwinmombay**

### CODE REVIEW

**Comment by erwinmombay** [comment]
@cramforce PTAL

File: build-system/tasks/changelog.js [hunk 1]
```diff
@@ -200,81 +268,150 @@ function buildSections(githubMetadata) {
...
-  };
+/**
+ * @param {!GitMetadataDef} gitMetadata
+ * @return {!Promise<!GitMetadataDef>}
+ */
+function getGithubPullRequestsMetadata(gitMetadata) {
+  // (erwinm): Github seems to only return data for the first 3 pages
+  // from my manual testing.
+  return BBPromise.all([
+      getClosedPullRequests(1),
```
Comment (cramforce) [inline build-system/tasks/changelog.js]: LOL, this will break in very productive weeks :)

**Comment by cramforce** [comment]
LGTM

File: build-system/tasks/changelog.js (hunk 1 above)
Comment (erwinmombay) [inline build-system/tasks/changelog.js]: haha yeah, it actually already did for one commit (wasn't in the first 3 pages). we do still fall back below to fetching /pulls/${id} if we can parse the id from the commit message and if nothing else just use the commit message w/o PR link. for some reason github returns no data with page 4 and up. (used curl)

**Pull request merged**